document/src/vespa/document/base/field.h

```diff
@@ -28,16 +28,35 @@ class Field final : public vespalib::FieldBase,
     const DataType *_dataType;
     int             _fieldId;
 public:
-    typedef std::shared_ptr<const Field> CSP;
-    typedef std::shared_ptr<Field> SP;
+    using CSP = std::shared_ptr<const Field>;
+    using SP = std::shared_ptr<Field>;
+    using CPtr = const Field *;
 
     struct FieldPtrComparator {
-        bool operator()(const Field* f1, const Field* f2) const {
+        bool operator()(CPtr f1, CPtr f2) const {
             return (*f1 < *f2);
         }
     };
 
-    using Set = std::set<const Field*, FieldPtrComparator>;
+    class Set {
+    public:
+        class Builder {
+        public:
+            Builder & reserve(size_t sz) { _vector.reserve(sz); return *this; }
+            Builder & insert(CPtr field) { _vector.emplace_back(field); return *this; }
+            Set build() { return Set(std::move(_vector)); }
+        private:
+            std::vector<CPtr> _vector;
+        };
+        bool contains(const Field & field) const;
```

Comment:
consider adding a contains taking a Field::Set as well since it can be implemented more efficiently using the std::includes function (compared to calling contains per field in the other set)

Reply:
Good idea, done.